src/load.ts

```diff
@@ -238,8 +238,9 @@ export default async function load(
       return;
     }
 
-    // hints help external schemas pick up where the root left off
-    const hint = getHint([...nodePath, ...ref.path], options.hint);
+    // hints help external partial schemas pick up where the root left off (for external complete/valid schemas, skip this)
+    const isRemoteFullSchema = ref.path[0] === "paths" || ref.path[0] === "components"; // if the initial ref is "paths" or "components" this must be a full schema
+    const hint = isRemoteFullSchema ? "OpenAPI3" : getHint([...nodePath, ...ref.path], options.hint);
```

Comment:
This was the erring code—given a path, we try and create a “hint” for any given subschema.

The reason hints are necessary is because the top-level keys of a schema can be anything, and there are no reserved words. For example, if `parameters` is a top-level property, is this subschema referring to properties on a SchemaObject? Or is this `#/components/parameters`? Or is this part of an operation object? We can only know by tracing the path from the root schema.

Of course, this whole process can be skipped for complete, valid schemas—and that’s exactly what this fix does.